tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -401,71 +479,74 @@ public void failCreateOnMismatchedDimensions() {
   }
 
   @Test
-  public void failCopyToOnIncompatibleDestination() {
+  public void whenCopyToWithDimensionMismatchThenThrowExpectedException() {
```

Comment:
Generally I aim to write test method names so that when they show up in a failure report, you don't have to look at the test code to work out what went wrong. The whenXThenY is a shorted version of behaviour driven development's Given When Then format.

In the case of failCopyToOnIncompatibleDestination() you know it should fail, but it doesn't tell you how its should fail, nor does it tell you what specific thing you were expecting to cause the failure. Anyone who has dug into the Tensor class will know that you can fail on a type mismatch, a dimension mismatch, or a shape mismatch, and all of these throw a IllegalArgumentException with different exception messages (or at least should do). 

We could change it to failCopyToOnIncompatibleDestinationShape() although that still doesn't tell us how its supposed to fail.